Chore/run setup from its

hozzjss opened the pull request.

hozzjss committed:
initial work for running setup from the service

hozzjss committed:
Add name, symbol, and decimals parameters to interchain token deployment functions

hozzjss committed:
Remove redundant contract-call for token setup in interchain token service implementation

hozzjss committed:
run setup from its

raress96 commented on contracts/native-interchain-token.clar:
```diff
@@ -366,18 +366,13 @@
 
 (define-public (setup
         (token-id_ (buff 32))
-        (token-type_ uint)
```
I suggest to leave the `token-type` and allow it to be either:
`TOKEN-TYPE-NATIVE-INTERCHAIN-TOKEN = 0`
or
`TOKEN-TYPE-MINT-BURN = 4`.

Currently ITS only supports `TOKEN-TYPE-NATIVE-INTERCHAIN-TOKEN`, but in the future after token linking is implemented it will also support `TOKEN-TYPE-MINT-BURN`. This way the NIT contract code can be forward compatible with what will be implemented next.

https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/interfaces/ITokenManagerType.sol#L15

hozzjss replied:
got it

raress96 commented on contracts/verify-onchain.clar:
```diff
@@ -1,6 +1,6 @@
-(define-constant nit-contract-code ";; title: native-interchain-token\n;; version:\n;; summary:\n;; description:\n\n;; traits\n;;\n(impl-trait 'ST1DRWEN0XDBPSEX3XVXK57NNT03B6ZZ0D58N36RS.traits.native-interchain-token-trait)\n(use-trait sip-010-trait 'ST1DRWEN0XDBPSEX3XVXK57NNT03B6ZZ0D58N36RS.traits.sip-010-trait)\n\n(define-constant ERR-NOT-AUTHORIZED (err u150000))\n(define-constant ERR-NON-STAN ...
+(define-constant nit-contract-code ";; title: native-interchain-token\n;; version:\n;; summary:\n;; description:\n\n;; traits\n;;\n(impl-trait 'ST1DRWEN0XDBPSEX3XVXK57NNT03B6ZZ0D58N36RS.traits.native-interchain-token-trait)\n(use-trait sip-010-trait 'ST1DRWEN0XDBPSEX3XVXK57NNT03B6ZZ0D58N36RS.traits.sip-010-trait)\n\n(define-constant ERR-NOT-AUTHORIZED (err u150000))\n(define-constant ERR-NON-STAN ...
```
Update the code after fix for the other comment is done.

raress96 commented on contracts/interchain-token-service-impl.clar:
```diff
@@ -687,6 +667,21 @@
+        ))
+
         ;; #[filter(verification-params, minter-unpacked, supply)]
         (try! (native-interchain-token-checks token minter-unpacked token-id supply
```
I think we no longer need the `native-interchain-token-checks` since it just checked token information that we already know now is correct.

raress96 commented:
One other thing I am thinking for the future token linking is to add an `execute-link-token` method that is not implemented yet in the proxy.

The signature will need to be mostly the same as `execute-deploy-interchain-token`, but I envision two cases possible here:
- linking with an existing token on Stacks -> lock/unlock TokenManager contract is needed
- linking with a new token on Stacks -> min/burn NativeInterchainToken contract is needed

So the signature should also include a token-manager trait, and the `native-interchain-token-trait` should be optional. Is that possible?

hozzjss committed:
feat: update setupNIT function to include tokenType parameter and adjust related tests

- Added tokenType parameter to setupNIT function in its-utils.ts with a default value of 0.
- Updated function arguments in verification-util.ts to pass tokenType when calling setupNIT.
- Modified mock parameters in nitMockParams and tmMockParams to reflect new token type and updated nonce values.
- Adjusted blockHeight and blockHeader values in mock parameters for consistency with recent changes.

hozzjss committed:
feat: implement execute-link-token function and add corresponding trait definition
